server/users/index.js

```diff
@@ -69,26 +69,32 @@ router.get('/:username/usermeta', async (req, res, next) => {
 			}
 
 			res.json(result);
+      next();
 		});
 	} catch (ex) {
 		next(ex);
 	}
-});
+};
+
+router.get('/:username/usermeta', router.getUsermetaMiddleware());
 
 
-router.put('/:username/usermeta/:key', async (req, res, next) => {
+router.updateMetaDataMiddleWare = (helperP = helpers) => async (req, res, next) => {
```

Comment:
`async` here is unnecessary, as it's not an `async` function (returning a `Promise`). General rule: only use `async` when you need to `await` things.

[This blog post here](https://medium.com/jspoint/javascript-promises-and-async-await-as-fast-as-possible-d7c8c8ff0abc) goes into details:
1. What's "Callback Hell"?
2. How Promises were introduced into JavaScript to save programmers from Callback Hell.
3. How `async` and `await` were later introduced as an improved way of using Promise, with a much less verbose syntax.

Note, I didn't read this blog post, but it seems like a like a fine explanation. Hopefully afterward you'll see why `async` is not necessary here ;-)

---

Same for the `getUsermetaMiddleware`.